api/src/main/java/com/budgettracker/api/models/UserCategory.java

```diff
@@ -0,0 +1,35 @@
+package com.budgettracker.api.models;
+
+import jakarta.persistence.*;
+import lombok.*;
+
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Size;
+import java.math.BigInteger;
+import java.time.LocalDateTime;
+
+@Getter
+@Setter
+@Entity(name="userCategory")
+@Table(name="userCategories")
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
```

Comment:
in order to soft detele please take a look at my discord message with soft deletion :

```
@SQLDelete(sql = "UPDATE entityName SET deleted = true WHERE id=?")
 //when doing repositoryName.delete(entity) -> will be just an update in db
@Where(clause = "deleted_at is NULL")  -> when selecting from table -> will be displayed only not-deleted entities
 ```


-> you have to check if this works for localdatetime field also, if not please update deletedAt in repository in order to separate the functionalities